From ec1e1101037fcd66967caa40bc2b038c928bae4f Mon Sep 17 00:00:00 2001
Subject: [PATCH] Bugfix for converting rounded rect to a d-string path

The previous algorithm supported the case where the rect was a python dictionary,
but not when it was a rect Element instance.
---
 svgpathtools/svg_to_paths.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/svgpathtools/svg_to_paths.py b/svgpathtools/svg_to_paths.py
index 65591af..c6ac6df 100644
--- a/svgpathtools/svg_to_paths.py
+++ b/svgpathtools/svg_to_paths.py
@@ -93,7 +93,8 @@ def rect2pathd(rect):
     rectangle object and proceed counter-clockwise."""
     x, y = float(rect.get('x', 0)), float(rect.get('y', 0))
     w, h = float(rect.get('width', 0)), float(rect.get('height', 0))
-    if 'rx' in rect or 'ry' in rect:
+
+    if 'rx' in rect.keys() or 'ry' in rect.keys():
 
         # if only one, rx or ry, is present, use that value for both
         # https://developer.mozilla.org/en-US/docs/Web/SVG/Element/rect
-- 
2.29.1

